src/prefect/tasks/secrets/env_var.py

```diff
@@ -0,0 +1,46 @@
+import os
+from typing import Any, Callable
+
+from prefect.tasks.secrets import Secret
+
+
+class EnvVarSecret(Secret):
+    """
+    A `Secret` task that retrieves a value from an environment variable.
+
+    Args:
+        - env_var (str): the environment variable that contains the secret value
+        - name (str, optional): a name for the task. If not provided, `env_var` is used.
+        - cast (Callable[[Any], Any]): A function that will be called on the Parameter
+            value to coerce it to a type.
+        - **kwargs (Any, optional): additional keyword arguments to pass to the Task constructor
+
+    Raises:
+        - ValueError: if a `result_handler` keyword is passed
+    """
+
+    def __init__(
+        self,
+        env_var: str,
+        name: str = None,
+        cast: Callable[[Any], Any] = None,
+        **kwargs
+    ):
+        self.env_var = env_var
+        self.cast = cast
+        if name is None:
+            name = env_var
+
+        super().__init__(name=name, **kwargs)
+
+    def run(self):
+        """
+        Returns the value of an environment variable after applying an optional `cast` function.
+
+        Returns:
+            - Any: the (optionally type-cast) value of the environment variable
+        """
+        value = os.getenv(self.env_var)
```

Comment:
Should we raise an error if the env var isn't found? That might prevent weird / hard-to-debug errors downstream

Reply:
I prefer the default behavior to be graceful since I think that's how most env var checks are typically handled but I added a kwarg for raising if it isn't set.